deps/querystring.lua

```diff
@@ -25,84 +25,89 @@ limitations under the License.
   tags = {"luvit", "url", "codec"}
 ]]
 
-local find = string.find
-local gsub = string.gsub
-local char = string.char
-local byte = string.byte
-local format = string.format
-local match = string.match
-local gmatch = string.gmatch
+
+local function hexToChar(hex)
+  return string.char(tonumber(hex, 16))
+end
+
+local function charToHex(character)
+  return string.format('%%%02X', string.byte(character))
+end
+
 
 local function urldecode(str)
-  str = gsub(str, '+', ' ')
-  str = gsub(str, '%%(%x%x)', function(h)
-    return char(tonumber(h, 16))
-  end)
-  str = gsub(str, '\r\n', '\n')
+  if str then
+    str = string.gsub(str, '+', ' ')
+    str = string.gsub(str, '%%(%x%x)', hexToChar)
+  end
   return str
 end
 
 local function urlencode(str)
   if str then
-    str = gsub(str, '\n', '\r\n')
-    str = gsub(str, '([^%w-_.~])', function(c)
-      return format('%%%02X', byte(c))
-    end)
+    str = string.gsub(str, '[^a-zA-Z0-9*%-%._]', charToHex)
   end
   return str
 end
 
-local function stringifyPrimitive(v)
-  return tostring(v)
-end
 
-local function stringify(params, sep, eq)
-  if not sep then sep = '&' end
-  if not eq then eq = '=' end
-  if type(params) == "table" then
-    local fields = {}
-    for key,value in pairs(params) do
-      local keyString = urlencode(stringifyPrimitive(key)) .. eq
-      if type(value) == "table" then
-        for _, v in ipairs(value) do
-          table.insert(fields, keyString .. urlencode(stringifyPrimitive(v)))
-        end
-      else
-        table.insert(fields, keyString .. urlencode(stringifyPrimitive(value)))
+local function stringify(tbl, sep, eq)
+  sep = sep or '&'
+  eq = eq or '='
+
+  if type(tbl) ~= 'table' then
```

Comment:
Move this above the `sep` and `eq` check.

Reply:
Alright, that's done.